contrib/endpoints/src/api_manager/api_manager_test.cc

```diff
@@ -83,6 +146,99 @@ TEST_F(ApiManagerTest, CorrectStatistics) {
   EXPECT_EQ(0, service_control_stat.send_report_operations);
 }
 
+// ApiManager instance should be initialized on ApiManager instance creation
+TEST_F(ApiManagerTest, InitializedOnApiManagerInstanceCreation) {
+  std::unique_ptr<MockApiManagerEnvironment> env(
+      new ::testing::NiceMock<MockApiManagerEnvironment>());
+
+  // ServiceManagement service should not be called
+  EXPECT_CALL(*(env.get()), DoRunHTTPRequest(_))
+      .Times(0)
+      .WillOnce(Invoke([this](HTTPRequest *req) {
+        EXPECT_TRUE(false);
+
+      }));
+
+  std::shared_ptr<ApiManager> api_manager(MakeApiManager(
+      std::move(env), kServiceConfig1, kServerConfigWithServiceNameConfigId));
+
+  // Already initialized
+  EXPECT_TRUE(api_manager);
+  EXPECT_EQ("OK", api_manager->ConfigLoadingStatus().ToString());
+
+  api_manager->Init();
+
+  // No change is expected
+  EXPECT_EQ("OK", api_manager->ConfigLoadingStatus().ToString());
```

Comment:
You can create api_manager_impl class and call its ConfigLoadingStatus() function.  The key is not to expose this function at api_manager interface.